.github/workflows/release.yml

```
@@ -1,76 +1,35 @@
-          java-version: 11
-          cache: sbt
-      - name: Release Snapshot (prerelease) to Sonatype
-        run: |
-          VERSION=$(git describe --tags | cut -f2 -d"@")
-          if [[ ${VERSION:0:1} == "v" ]] ; then
-            VERSION=${VERSION:1}
-          fi
-          if [[ ${VERSION: -9} != "-SNAPSHOT" ]] ; then
-            echo "Version must end in -SNAPSHOT. Adding -SNAPSHOT suffix"
-            VERSION="$VERSION-SNAPSHOT"
-          fi
-          echo $PGP_SECRET | base64 --decode | gpg --batch --import
-          export GPG_TTY=$(tty)
-          echo "Releasing version $VERSION Sonatype as snapshot"
-          yes | sbt -DRELEASE_TYPE=snapshot "clean" "release cross release-version $VERSION with-defaults"
-        env:
-          PGP_SECRET: ${{ secrets.PGP_SECRET }}
-          PGP_PASSPHRASE: ${{ secrets.PGP_PASSPHRASE }}
-          SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
-          SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
+  scala-maven-release:
+    name: Maven Release
+    uses: guardian/gha-scala-library-release-workflow/.github/workflows/reusable-release.yml@main
+    permissions: { contents: write, pull-requests: write }
+    secrets:
+      SONATYPE_PASSWORD: ${{ secrets.AUTOMATED_MAVEN_RELEASE_SONATYPE_PASSWORD }}
+      PGP_PRIVATE_KEY: ${{ secrets.AUTOMATED_MAVEN_RELEASE_PGP_SECRET }}
 
-  release_production_sonatype:
-    if: "! github.event.release.prerelease"
+  typescript-npm-release:
+    name: NPM Release
+    needs: scala-maven-release
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-        with:
-          base: main  #see https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#pull-request-events
-      - uses: actions/setup-java@v3
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
         with:
           distribution: corretto
           java-version: 11
           cache: sbt
-      - uses: actions/setup-node@v3
+      - uses: actions/setup-node@v4
         with:
           node-version-file: .nvmrc
           registry-url: https://registry.npmjs.org
-      - name: Release Production to Sonatype and NPM
+      - name: Release Typescript to NPM
         run: |
-          VERSION=$(git describe --tags | cut -f2 -d"@")
-          if [[ ${VERSION:0:1} == "v" ]] ; then
-            VERSION=${VERSION:1}
-          fi
-          if [[ ${VERSION: -9} == "-SNAPSHOT" ]] ; then
-            echo "Version must NOT end in -SNAPSHOT."
-            exit 1
-          fi
-          echo $PGP_SECRET | base64 --decode | gpg --batch --import
-          export GPG_TTY=$(tty)
-          echo "Releasing version $VERSION Sonatype as production"
-
-          yes | sbt -DRELEASE_TYPE=production "clean" "release cross release-version $VERSION with-defaults" "project typescriptClasses" "releaseNpm $VERSION"
+          sbt "project typescript" "releaseNpm ${{ needs.scala-maven-release.outputs.RELEASE_VERSION }}"
```

I think this repo has slightly different typescript sub-project naming to `content-api-models` (where the sub-project [_is_](https://github.com/guardian/content-api-models/blob/d77f319fe0572137d862dfb2df9911b6d49dc67a/build.sbt#L142) just called `typescript`), so we need this:

```suggestion
          sbt "project typescriptClasses" "releaseNpm ${{ needs.scala-maven-release.outputs.RELEASE_VERSION }}"
```

...see:

https://github.com/guardian/content-atom/blob/8a8b60cf26a0d8cf75e849cef36c7308a35095f8/build.sbt#L148

It would be nice if both projects had a uniform layout and naming, but that might be for another PR!

---

Very well spotted, it's making more sense to me now.
I will change the typescript project name.